### DISCUSSION TITLE
GUI Images

### DISCUSSION THREAD

**Discussion by SebGue** [Eomys/pyleecan#326] [Q&A] [answered]
Hello @BonneelP , 

are there some script files, that generated the GUI images (e.g. of the machine types). If they are not available yet, would you mind to commit them?

Best regards, Sebastian

**Comment by BonneelP** [comment, marked as answer]
Hello Sebastian,

All images of the GUI are generated in the scripts from Tests/Plot/Schematics. For the Step 1: Machine Type, the images are generated with Tests/Plot/Schematics/test_machine_img but indeed for now it only plot SCIM, IPMSM, SynRM, SPMSM. The automatization is new, before it was just some screenshots. I only added these machines because they are the one available in Manatee. The missing machines are maybe in pyleecan/Data/Machine otherwise they may need to be defined again. Are you looking for an image in particular? 

Best regards,
Pierre

**Reply by SebGue** [reply]
The Maschine images have different image sizes and the SRM image is wrong, i think. I wanted to fix that

**Reply by BonneelP** [reply]
Yes before it was all screenshot and we updated only a part of the images with the machines that were available in Data. The best solution to fix that would be to provide a "reference machine" (as Prius is for IPMSM) for each topology (ideally from a publication) and then use corresponding plot as the default image by extending Tests/Plot/Schematics/test_machine_img to normalize the images.

**Reply by SebGue** [reply]
I think we should move this to the generator to automatically update the resources. This would also enable for easy change in color schemes ;-)
